awx/ui/client/lib/components/syntax-highlight/syntax-highlight.directive.js

```diff
@@ -0,0 +1,89 @@
+const templateUrl = require('~components/syntax-highlight/syntax-highlight.partial.html');
+
+function atSyntaxHighlightController ($scope, AngularCodeMirror) {
+    const vm = this;
+    const varName = `${$scope.name}_codemirror`;
+
+    function init () {
+        if ($scope.disabled === 'true') {
```

Comment:
I'm having a little bit of trouble tracing this part of the code and I can't quite remember how the ngDisabled directive works.  I only see one place where the atSyntaxHighlight directive is used:

```
                        html += '<at-syntax-highlight ';
                        html += `id="${form.name}_${fld}" `;
                        html += `class="${field.class}" `;
                        html += `label="${field.label}" `;
                        html += `tooltip="${field.awPopOver || ''}" `;
                        html += `name="${fld}" `;
                        html += `value="${fld}" `;
                        html += `default="${field.default || ''}" `;
                        html += `rows="${field.rows || 6}" `;
                        html += `mode="${field.mode}" `;
                        html += `ng-disabled="${field.ngDisabled}" `;
                        html += '></at-syntax-highlight>';
```

It looks like we pass through ngDisabled from form field.  I guess I'm trying to figure out why it comes through as a string and not a bool?

Reply:
I'm not exactly sure either honestly. I copied this portion from the code mirror directive